runtime/filetype.vim

```diff
@@ -1144,7 +1144,7 @@ au BufNewFile,BufRead *.hook
 	\ endif
 
 " Pam conf
-au BufNewFile,BufRead */etc/pam.conf			setf pamconf
+au BufNewFile,BufRead */etc/pam.conf			let pamconf_has_service_field = 1 | setf pamconf
```

Comment:
```suggestion
au BufNewFile,BufRead */etc/pam.conf			let b:pamconf_has_service_field = 1 | setf pamconf
```
Using a buffer local variable might be better.
For example, like this part:
https://github.com/vim/vim/blob/77cae09642353c625ccedd6a8a691a700e5e6315/runtime/filetype.vim#L1250-L1253